stonesoup/simulator/transition.py

```diff
@@ -0,0 +1,266 @@
+# -*- coding: utf-8 -*-
+from datetime import timedelta
+from copy import deepcopy
+
+import numpy as np
+
+from stonesoup.models.transition.linear import ConstantTurn, ConstantVelocity,\
+    CombinedLinearGaussianTransitionModel
+from stonesoup.types.array import StateVector
+
+
+def get_smooth_transition_models(initial_state, x_coords, y_coords, times, turn_rate):
+    """Generate a list of constant-turn and constant acceleration transition models alongside a
+    list of transition times to provide smooth transitions between 2D cartesian coordinates and
+    time pairs.
+
+    Parameters
+    ----------
+    initial_state: :class:`~.State` The initial state of the platform.
+    x_coords: A list of int/float x-coordinates (cartesian) in the order that the target must
+        follow.
+    y_coords: A list of int/float y-coordinates (cartesian) in the order that the target must
+        follow.
+    times: A list of :class:`~.datetime.datetime` dictating the times at which the target
+        must be at each corresponding coordinate.
+    turn_rate: Float
+        Angular turn rate (radians/second) measured anti-clockwise from positive x-axis.
+
+    Returns
+    -------
+    transition_models: A list of :class:`~.ConstantTurn` and
+        :class:`~.Point2PointConstantAcceleration` transition models.
+    transition_times: A list of :class:`~.datetime.timedelta` dictating the transition time
+        for each corresponding transition model in transition_models.
+
+    Notes
+    -----
+    x_coords, y_coords and times must be of same length.
+    """
+
+    state = deepcopy(initial_state)  # don't alter platform state with calculations
+
+    if len(x_coords) != len(y_coords) or len(x_coords) != len(times):
+        raise ValueError('x_coords, y_coords and times must be same length')
+
+    transition_models = []
+    transition_times = []
+
+    for i in range(1, len(times)):  # make models to reach next coordinates from previous.
+
+        dx = x_coords[i] - state.state_vector[0]  # distance to next x-coord
+        dy = y_coords[i] - state.state_vector[2]  # distance to next y-coord
+
+        if i == 0 and dx == 0 and dy == 0:
+            a = 0  # if initial and second target coordinates are same, set arbitrary bearing of 0
+
+        vx = state.state_vector[1]  # initial x-speed
+        vy = state.state_vector[3]  # initial y-speed
+
+        if vx != 0 or vy != 0:  # if velocity is 0, keep previous bearing
+            a = np.arctan2(vy, vx)  # initial bearing
+
+        if dx == 0 and dy == 0 and vx == 0 and vy == 0:  # if at destination with 0 speed, stay
+            transition_times.append(times[i])
+            transition_models.append(CombinedLinearGaussianTransitionModel((ConstantVelocity(0),
+                                                                            ConstantVelocity(0))))
+            continue
+
+        d = np.sqrt(dx ** 2 + dy ** 2)  # distance to next coord
```

Comment:
This would be clear for calculations, grouping by operator order.
```python
np.sqrt(dx**2 + dy**2)
```

Other places in PR could be changed as well.